runtime/near-vm-runner/src/instrument/stack_height/thunk.rs

```diff
@@ -3,7 +3,7 @@ use parity_wasm::{
     builder,
     elements::{self, FunctionType, Internal},
 };
-use std::collections::HashMap as Map;
```

Comment:
I think we should create an issue to review all uses of `HashMap`, `HashSet`, etc. in our code to ensure that there aren't any non-determinism issues there.

Reply:
FWIW, I'd be curious to hear some case studies of how to best enforce determinism consistently. 

* The simplest option is indeed to use tree-based collections, but my understanding is that it is pretty inefficient, as collection lookups are usually hot in profiles, and trees are way less CPU&memory efficient than hashmaps. No gut feeling for just *how* inefficient it is.
* The second simplest option is to use hash-maps with deterministic iteration (index map, what modern python is doing). Again, this is trading perf for determinism, but I don't know how much perf. I *think* this would probably be my default pick for making deterministic software. 
* The third option is to make sure that `HashMap` doesn't expose non-deterministic API (doesn't implement `iter`). I think this is what `rustc` is doing, not sure how successful it is. 

Interestingly, I *think* in `rust-analyzer` we've never ever hit this problem. This is surprising to me, as, given our incremental compilation setup, the non-determinism would lead to straightforward panic, so it'd be hard to miss. I *think* that is because in rust-analyzer we are using `FxHashMap` which doesn't do randomization (so, DOS-vulnerable).